pkg/internal/checkup/checkup_test.go

```diff
@@ -68,42 +72,168 @@ func TestCheckupShouldSucceed(t *testing.T) {
 	assert.Empty(t, testClient.createdVMIs)
 
 	expectedResults := successfulRunResults(vmiUnderTestName)
-	actualResults := testCheckup.Results()
+	actualResults := reporter.FormatResults(testCheckup.Results())
 	assert.Equal(t, expectedResults, actualResults)
 }
 
+func TestCheckupShouldReturnErrorWhen(t *testing.T) {
+	tests := map[string]struct {
+		clientConfig    clientConfig
+		expectedResults map[string]string
+		expectedErr     string
+	}{
+		"noStorageClasses": {
+			clientConfig:    clientConfig{noStorageClasses: true},
+			expectedResults: map[string]string{reporter.DefaultStorageClassKey: checkup.ErrNoDefaultStorageClass},
+			expectedErr:     checkup.ErrNoDefaultStorageClass,
+		},
+		"noDefaultStorageClass": {
+			clientConfig:    clientConfig{noDefaultStorageClass: true},
+			expectedResults: map[string]string{reporter.DefaultStorageClassKey: checkup.ErrNoDefaultStorageClass},
+			expectedErr:     checkup.ErrNoDefaultStorageClass,
+		},
+		"multipleDefaultStorageClasses": {
+			clientConfig:    clientConfig{multipleDefaultStorageClasses: true},
+			expectedResults: map[string]string{reporter.DefaultStorageClassKey: checkup.ErrMultipleDefaultStorageClasses},
+			expectedErr:     checkup.ErrMultipleDefaultStorageClasses,
+		},
+		"storageProfileIncomplete": {
+			clientConfig:    clientConfig{spIncomplete: true},
+			expectedResults: map[string]string{reporter.StorageProfilesWithEmptyClaimPropertySetsKey: testScName, reporter.StorageProfilesWithSpecClaimPropertySetsKey: testScName, reporter.StorageWithRWXKey: ""},
```

Comment:
Not sure how can a SC be on both EmptyClaimPropertySets and SpecClaimPropertySets? I guess only in UT it's possible?